mm2src/mm2_main/src/lp_wallet/mnemonics_storage.rs

```diff
@@ -64,8 +64,17 @@ pub(super) async fn read_encrypted_passphrase_if_available(ctx: &MmArc) -> Walle
 }
 
 pub(super) async fn read_all_wallet_names(ctx: &MmArc) -> WalletsStorageResult<impl Iterator<Item = String>> {
-    let wallet_names = list_files_by_extension(&ctx.db_root(), "dat", false)
+    let ext: &str = "dat";
+    let wallet_names = list_files_by_extension(&ctx.db_root(), ext, false)
```

Comment:
I think `list_files_by_extension` should handle this logic as it stands as "list files", so it shouldn't return directories.

Also, it's better to use `is_dir` with `!` as `is_file` expects path to be present on disk.

Reply:
Logic moved into `filter_files_by_extension`.